File: DevExpress-Demo/CS/EditorsDemo.Wpf/ViewModels/LinearBarCodeViewModel.cs
using DevExpress.XtraPrinting.BarCode.Native;
using System;
using System.Collections.Generic;
using System.Linq;

namespace EditorsDemo {
    public class LinearBarCodeViewModel {
        public LinearBarCodeViewModel() {
            this.BarCodeSymbology = BarCodeSymbology.Code128;
            this.Text = "101";
            this.BarCodeModule = 3;
            this.AutoModule = true;
            this.ShowText = true;
        }
        public virtual bool ShowText { get; set; }
        public virtual bool AutoModule { get; set; }
        public virtual string Text { get; set; }
        public virtual double BarCodeModule { get; set; }
        public virtual BarCodeSymbology BarCodeSymbology { get; set; }
        public IEnumerable<BarCodeSymbology> BarCodeTypes { get { return GetBarCodeTypes(); } }
        protected virtual IEnumerable<BarCodeSymbology> GetBarCodeTypes() {
            return Enum.GetValues(typeof(BarCodeSymbology)).Cast<BarCodeSymbology>().TakeWhile(bcs =>
                bcs != BarCodeSymbology.QRCode &&
                bcs != BarCodeSymbology.DataMatrix &&
                bcs != BarCodeSymbology.DataMatrixGS1 &&
                bcs != BarCodeSymbology.PDF417);
        }
    }
    public class BarCode2DViewModel : LinearBarCodeViewModel {
        public BarCode2DViewModel() {
            this.BarCodeSymbology = BarCodeSymbology.QRCode;
        }
        protected override IEnumerable<BarCodeSymbology> GetBarCodeTypes() {
            return new BarCodeSymbology[] { BarCodeSymbology.QRCode, BarCodeSymbology.DataMatrix, BarCodeSymbology.DataMatrixGS1, BarCodeSymbology.PDF417 };
        }
    }
}

File: DevExpress-Demo/VB/EditorsDemo.Wpf/ViewModels/LinearBarCodeViewModel.vb
Imports DevExpress.XtraPrinting.BarCode.Native
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace EditorsDemo

    Public Class LinearBarCodeViewModel

        Public Sub New()
            BarCodeSymbology = BarCodeSymbology.Code128
            Text = "101"
            BarCodeModule = 3
            AutoModule = True
            ShowText = True
        End Sub

        Public Overridable Property ShowText As Boolean

        Public Overridable Property AutoModule As Boolean

        Public Overridable Property Text As String

        Public Overridable Property BarCodeModule As Double

        Public Overridable Property BarCodeSymbology As BarCodeSymbology

        Public ReadOnly Property BarCodeTypes As IEnumerable(Of BarCodeSymbology)
            Get
                Return GetBarCodeTypes()
            End Get
        End Property

        Protected Overridable Function GetBarCodeTypes() As IEnumerable(Of BarCodeSymbology)
            Return [Enum].GetValues(GetType(BarCodeSymbology)).Cast(Of BarCodeSymbology)().TakeWhile(Function(bcs) bcs <> BarCodeSymbology.QRCode AndAlso bcs <> BarCodeSymbology.DataMatrix AndAlso bcs <> BarCodeSymbology.DataMatrixGS1 AndAlso bcs <> BarCodeSymbology.PDF417)
        End Function
    End Class

    Public Class BarCode2DViewModel
        Inherits LinearBarCodeViewModel

        Public Sub New()
            BarCodeSymbology = BarCodeSymbology.QRCode
        End Sub

        Protected Overrides Function GetBarCodeTypes() As IEnumerable(Of BarCodeSymbology)
            Return New BarCodeSymbology() {BarCodeSymbology.QRCode, BarCodeSymbology.DataMatrix, BarCodeSymbology.DataMatrixGS1, BarCodeSymbology.PDF417}
        End Function
    End Class
End Namespace
